### PR TITLE
Improve testing environment for whole repo

### PR DESCRIPTION
**Pull request by jakubzaba**
Add circleci
Move enzyme into root package.json
Add testing scripts into root package.json
Turn off some eslint rules

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -9,10 +13,24 @@
   ],
   "plugins": [
     "flowtype",
-    "prettier"
+    "prettier",
+    "jest"
   ],
   "rules": {
-    "prettier/prettier": "error",
+    "prettier/prettier": [
```
Comment (oreqizer) [inline .eslintrc]: nope, use `.prettierrc`. also, the only rules that make sense configuring are the ones already present

File: .eslintrc [hunk 2]
```diff
@@ -9,10 +13,24 @@
...
+      "error",
+      {
+        "bracketSpacing": true,
+        "printWidth": 80,
+        "singleQuote": false,
+        "tabWidth": 2,
+        "trailingComma": "all"
+      }
+    ],
+    "import/no-extraneous-dependencies": "off",
```
Comment (oreqizer) [inline .eslintrc]: nope

File: .eslintrc [hunk 3]
```diff
@@ -9,10 +13,24 @@
...
+    "react/jsx-filename-extension": "off",
```
Comment (oreqizer) [inline .eslintrc]: why?

File: .eslintrc [hunk 4]
```diff
@@ -9,10 +13,24 @@
...
+    "react/require-default-props": "off",
```
Comment (oreqizer) [inline .eslintrc]: why?

File: .eslintrc [hunk 5]
```diff
@@ -9,10 +13,24 @@
...
+    "jsx-a11y/label-has-for": "off",
```
Comment (oreqizer) [inline .eslintrc]: why?

File: .eslintrc (hunk 2 above)
Comment (Fallion) [reply .eslintrc]: I agree with @oreqizer . If this is causing issues in tests then use a glob pattern to fix it. https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-extraneous-dependencies.md

File: .eslintrc (hunk 3 above)
Comment (jaroslav-kubicek) [reply .eslintrc]: I would agree with turning this off, I don't see any reason to make difference whether file has some JSX or not, it's like prefixing all string variables with `s`...

File: .eslintrc (hunk 2 above)
Comment (jakubzaba) [reply .eslintrc]: It's not working for yarn workspaces. Globs didn't helped for all cases.
https://github.com/benmosher/eslint-plugin-import/issues/458#issuecomment-361352578

File: .eslintrc (hunk 3 above)
Comment (jakubzaba) [reply .eslintrc]: Changed to `.js` only. I don't see benefit to use `.jsx`.

File: .eslintrc (hunk 4 above)
Comment (jakubzaba) [reply .eslintrc]: removed

File: .eslintrc (hunk 5 above)
Comment (jakubzaba) [reply .eslintrc]: Added comment. Label is wrapping input, no need for `for`.

File: .eslintrc (hunk 1 above)
Comment (jakubzaba) [reply .eslintrc]: Reverted

**Comment by jakubzaba** [comment]
@oreqizer pls take a look on changes.

**Comment by jakubzaba** [comment]
To push this forward I suggest to merge it. All comments were fixed or answered.
@Fallion @oreqizer

File: .eslintrc [hunk 6]
```diff
@@ -9,10 +13,14 @@
   ],
   "plugins": [
     "flowtype",
-    "prettier"
+    "prettier",
+    "jest"
   ],
   "rules": {
     "prettier/prettier": "error",
+    "import/no-extraneous-dependencies": "off",
```
Comment (oreqizer) [inline .eslintrc]: no

File: .eslintrc (hunk 3 above)
Comment (oreqizer) [reply .eslintrc]: well JSX is not a valid JS syntax, that's a pretty simple reason. whatever, you guys are the majority

File: .eslintrc (hunk 2 above)
Comment (oreqizer) [reply .eslintrc]: so turn off where needed

**Comment by oreqizer** [comment]
sorry for the delay, guys. yesterday was hectic

**Comment by jakubzaba** [comment]
I added `devDependencies` for `import/no-extraneous-dependencies` and `react/require-default-props` because workaround for Flow would be insane.

**Comment by oreqizer** [comment]
The point about working around Flow is invalid IMO, but OK, requiring all props to be defaulted would be daunting I guess

**Pull request merged**